Jellyfin.Server.Implementations/Item/BaseItemRepository.cs

```diff
@@ -506,59 +507,98 @@ public void UpdateOrInsertItems(IReadOnlyList<BaseItemDto> items, CancellationTo
                 context.BaseItemProviders.Where(e => e.ItemId == entity.Id).ExecuteDelete();
                 context.BaseItems.Attach(entity).State = EntityState.Modified;
             }
+        }
 
-            context.AncestorIds.Where(e => e.ItemId == entity.Id).ExecuteDelete();
-            if (item.Item.SupportsAncestors && item.AncestorIds != null)
+        context.SaveChanges();
```

Comment:
Why are these added? To fail early? Is it even necessary to use a transaction here?

Reply:
in my tests this seem to help with both performance and possible blocking for transactions